core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -38,207 +40,221 @@
-import alluxio.resource.AlluxioResourceLeakDetectorFactory;
-import alluxio.retry.RetryPolicy;
-import alluxio.retry.RetryUtils;
-import alluxio.security.user.UserState;
+import alluxio.retry.CountingRetry;
 
-import com.google.common.io.Closer;
 import com.google.common.util.concurrent.ListenableFuture;
-import io.grpc.StatusRuntimeException;
 import io.grpc.stub.StreamObserver;
-import io.netty.util.ResourceLeakDetector;
-import io.netty.util.ResourceLeakTracker;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
 import java.util.concurrent.TimeUnit;
-import javax.annotation.Nullable;
 
 /**
  * Default implementation of {@link BlockWorkerClient}.
  */
-public class DefaultBlockWorkerClient implements BlockWorkerClient {
+public class DefaultBlockWorkerClient extends AbstractClient implements BlockWorkerClient {
   private static final Logger LOG =
       LoggerFactory.getLogger(DefaultBlockWorkerClient.class.getName());
 
-  private static final ResourceLeakDetector<DefaultBlockWorkerClient> DETECTOR =
-      AlluxioResourceLeakDetectorFactory.instance()
-          .newResourceLeakDetector(DefaultBlockWorkerClient.class);
-
+  /**Streaming channel used for streaming rpc calls. */
   private GrpcChannel mStreamingChannel;
-  private GrpcChannel mRpcChannel;
   private final GrpcServerAddress mAddress;
   private final long mRpcTimeoutMs;
 
-  private final BlockWorkerGrpc.BlockWorkerStub mStreamingAsyncStub;
-  private final BlockWorkerGrpc.BlockWorkerBlockingStub mRpcBlockingStub;
-  private final BlockWorkerGrpc.BlockWorkerFutureStub mRpcFutureStub;
-
-  @Nullable
-  private final ResourceLeakTracker<DefaultBlockWorkerClient> mTracker;
+  private BlockWorkerGrpc.BlockWorkerStub mStreamingAsyncStub;
+  private BlockWorkerGrpc.BlockWorkerBlockingStub mRpcBlockingStub;
+  private BlockWorkerGrpc.BlockWorkerFutureStub mRpcFutureStub;
 
   /**
    * Creates a client instance for communicating with block worker.
    *
-   * @param userState     the user state
+   * @param context     the client context
    * @param address     the address of the worker
    * @param alluxioConf Alluxio configuration
    */
-  public DefaultBlockWorkerClient(UserState userState, GrpcServerAddress address,
-      AlluxioConfiguration alluxioConf) throws IOException {
-    RetryPolicy retryPolicy = RetryUtils.defaultClientRetry();
-    UnauthenticatedException lastException = null;
-    // TODO(feng): unify worker client with AbstractClient
-    while (retryPolicy.attempt()) {
-      try {
-        // Disables channel pooling for data streaming to achieve better throughput.
-        // Channel is still reused due to client pooling.
-        mStreamingChannel = GrpcChannelBuilder.newBuilder(address, alluxioConf)
-            .setSubject(userState.getSubject())
+  public DefaultBlockWorkerClient(ClientContext context, GrpcServerAddress address,
+                                  AlluxioConfiguration alluxioConf) {
```

Comment:
`alluxioConf` can be obtained from `ClientContext`, so it's redundant.

Reply:
Fixed